Moq.AutoMock/MockArrayInstance.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Moq.AutoMock
+{
+    internal sealed class MockArrayInstance : IInstance
+    {
+        private readonly Type _type;
+        private readonly List<IInstance> _mocks = new List<IInstance>();
+
+        public MockArrayInstance(Type type) => _type = type;
+
+        public object Value
+        {
+            get
+            {
+                int i = 0;
+                Array array = Array.CreateInstance(_type, _mocks.Count);
+                foreach (IInstance instance in _mocks)
```

Comment:
`((IList)_mocks).CopyTo(array);`

Reply:
That doesn't work because we're actually setting the `.Value` into the array, not the instance.

Reply:
Ah I missed that.